src/Microsoft.DotNet.Watcher.Core/Internal/ProcessExtensions.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using Microsoft.Extensions.PlatformAbstractions;
+
+namespace Microsoft.DotNet.Watcher.Core.Internal
+{
+    public static class ProcessExtensions
+    {
+        private static readonly bool _isWindows = PlatformServices.Default.Runtime.OperatingSystemPlatform == Platform.Windows;
+
+        public static void KillTree(this Process process, TimeSpan timeout)
+        {
+            string stdout;
+            if (_isWindows)
+            {
+                RunProcessAndWaitForExit(
+                    "taskkill",
+                    $"/T /F /PID {process.Id}",
+                    timeout,
+                    out stdout);
+            }
+            else
+            {
+                var childs = GetAllChildIdsUnix(process, timeout);
+                foreach (var childId in childs)
+                {
+                    KillProcessUnix(childId, timeout);
```

Comment:
Do you kill the original process? Seems like you are only killing descendents of the original process.